values.yaml

```diff
@@ -62,6 +62,17 @@ st2:
     [api]
     allow_origin = '*'
 
+  # Optional configuration for handling a precomputed datastore encryption key
+  datastoreEncryption:
+    enabled: false
+    # Where will the key be mounted in the containers
+    encryptionKeyPath: /etc/st2/keys
+  # secret:
+  #   # The name of the secret in which the key is stored
+  #   name: st2-encryption-key
+  #   # The key containing the json formated encryption key
+  #   key: datastore_encryption_key.json
```

Comment:
For simplicity reasons, I'd prefer only the following Helm value to be exposed to the user:

```yaml
secrets:
  st2:
    # ST2 crypto key for the  K/V datastore. 
    # See https://docs.stackstorm.com/datastore.html#securing-secrets-admin-only for more info.
    # Warning! Replace with your own generated key!
    #datastore_crypto_key: {"hmacKey": {"hmacKeyString": "", "size": 256}, "size": 256, "aesKeyString": "", "mode": "CBC"}    
```

And so if `secrets.st2.datastore_crypto_key` value is present, st2 crypto key for the datastore will be enabled automatically with all the K8s secrets under the hood and defaults like https://github.com/StackStorm/st2-packages/blob/e86adb5e75b5013dac43198f9e464b29cb08ed58/scripts/st2bootstrap-deb.sh#L335-L361


See for example other secrets like https://github.com/StackStorm/stackstorm-ha/blob/master/templates/secrets_ssh.yaml for more mechanics.

With that, crypto key path would be `/etc/st2/keys/datastore_key.json`.
Services that need shared access are `st2api st2sensorcontainer st2workflowengine st2actionrunner` and `st2client` pods.

Reply:
I've implemented as you advised, it's way less complex for the end user indeed.